payload/reggae/build.d

```diff
@@ -598,9 +598,9 @@ struct Command {
 
             case link: {
                 version(Windows)
-                    auto cArgs = ["/nologo", "/Fe$out", "$flags", "$in"];
+                    auto cArgs = ["/nologo", "/Fe$out", "$flags", "$link_libraries", "$in"];
                 else
-                    auto cArgs = ["-o", "$out", "$flags", "$in"];
+                    auto cArgs = ["$flags", "$in", "-o", "$out", "$link_libraries"];
```

Comment:
Why the change in order?

Reply:
Basically, the answer is "that's how Ninja does it". 

With the _old_ order, the problem that I encountered had to do with the fact that the static library was _not_ the last thing in the command line, e.g. `gcc -o binary flag1 flag2 libStatic.a main.o`. When this happened, it did not link correctly and complained that the symbols referenced by `main.o` couldn't be found. Note that on my machine (I'm using Ubuntu 20) the test was passing and there were no problems. However, on these upstream Linux runners (that use Ubuntu 22) the test was failing, so I changed the order and it worked.

As I said, that's also how Ninja does it. And, in general, what I've seen is that Ninja makes a lot of distinctions when it comes to `flags`, which reggae doesn't when it generates the `build.ninja`  as it just dumps everything under the same `flags` tag and is not interested in the order of the flags at all. On the other hand, Ninja makes all these distinctions between the flags essentially to pass everything in the right order.

However, for the moment, I think what we have suits our needs.